surfactant/filetypeid/id_magic.py

```diff
@@ -28,7 +28,22 @@ def identify_file_type(filepath: str) -> Optional[str]:
             if magic_bytes[:4] == b"\x7fELF":
                 return "ELF"
             if magic_bytes[:2] == b"MZ":
+                # Check to see if this is actually a PE file or a DOS file
+                coff_addr = (
+                    int.from_bytes(magic_bytes[0x3C:0x40], byteorder="little", signed=False)
+                    & 0xFFFF
+                )
+                if magic_bytes[coff_addr : coff_addr + 2] != b"PE":
+                    return "DOS"
+
+                # Check for the linux kernel header at 0x202 (requires a second read)
+                magic_bytes = f.read(265)
+                if magic_bytes[249:253] == b"HdrS":
```

Comment:
```suggestion
                magic_bytes += f.read(265)
                if magic_bytes[0x202:0x206] == b"HdrS":
```

Supporting file checks that require accessing data past the initial read is something I've been meaning to figure out -- I think for the subsequent read appending the new data to the previous bytes read could make it easier to follow what offset in the file is being checked. Something similar could be done for ensuring enough data is read when the offset to check if specified by another field in the header (e.g. `coff_addr`).

Reply:
Definitely, that's something that I was thinking about, but wasn't sure how you all would like to best handle that. I think that this looks good -- I just used another 265 byte read as that was the same size as the initial read.